test/controllers/concerns/parameters/user_test.rb

```diff
@@ -34,6 +39,27 @@ class UserParametersTest < ActiveSupport::TestCase
       params = ActionController::Parameters.new(:user => {:login => 'another'})
       assert_includes(as_admin { self.class.user_params_filter.filter_params(params, context) }, 'login')
     end
+
+    test "permits login attribute for external users with permissions" do
+      params = ActionController::Parameters.new(:user => {:login => 'another'})
+      as_user (FactoryBot.create(:user, :auth_source => @auth_source_ldap, :roles => [roles(:manager)])) do
```

Comment:
`.build` or `.build_stubbed` is probably good enough in every new test here. That would only create the object in memory rather than making the roundtrip to the database & commit (so it'd be faster).